src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/resources/viewmodel/ResourcesViewModel.kt

```diff
@@ -0,0 +1,46 @@
+package org.wycliffeassociates.otter.jvm.app.ui.resources.viewmodel
+
+import io.reactivex.Observable
+import javafx.application.Platform
+import javafx.beans.property.SimpleObjectProperty
+import javafx.beans.property.SimpleStringProperty
+import org.wycliffeassociates.otter.common.data.workbook.*
+import org.wycliffeassociates.otter.jvm.app.widgets.resourcecard.model.ResourceGroupCardItemList
+import org.wycliffeassociates.otter.jvm.app.widgets.resourcecard.model.resourceGroupCardItem
+import tornadofx.*
+
+class ResourcesViewModel : ViewModel() {
+    val activeWorkbookProperty = SimpleObjectProperty<Workbook>()
+    val workbook: Workbook
+        get() = activeWorkbookProperty.value
+
+    val activeChapterProperty = SimpleObjectProperty<Chapter>()
+    val chapter: Chapter
+        get() = activeChapterProperty.value
+
+    val activeResourceSlugProperty = SimpleStringProperty()
+    val resourceSlug: String
+        get() = activeResourceSlugProperty.value
+
+    val resourceGroups: ResourceGroupCardItemList = ResourceGroupCardItemList(mutableListOf())
+
+    fun <T, R:Any> Observable<T>.mapNotNull(f: (T) -> R?): Observable<R> =
```

Comment:
this should probably just be moved out to its own file in a utils package

Reply:
Where would you like to see this go? Should a utils package be made in common or does it need to be under another package such as observable > utils?

Reply:
put it in common.utils

maybe like RxUtils.kt

Reply:
done

Reply:
@KJoslyn If it's a util, I guess it should have kdoc:
```
/**
 * Apply the given function as with `map()`, but remove any resulting nulls from the
 * observable stream.
 * 
 * This is how `observable.map(f).filter { it != null }` should behave, but it keeps
 * nulls from ever appearing in an Observable, which would cause a crash in that
 * two-step map/filter example.
 */
```

Reply:
Awesome, thanks! Just added that.